linkcheck/HtmlParser/htmlsax.py

```diff
@@ -27,6 +27,10 @@
 from bs4 import BeautifulSoup, Tag
 
 
+def make_soup(markup, from_encoding=None):
```

Comment:
is it okay to have multiple such objects in parallel? AKA should this be a singleton?

Reply:
I think we want multiple objects - they are not just interfaces to a parser but representations of that piece of markup with methods you can use to interrogate.